plugins/ftdidmx/FtdiWidget.cpp

```diff
@@ -364,14 +372,35 @@ bool FtdiInterface::Write(const ola::DmxBuffer& data) {
   }
 }
 
-bool FtdiInterface::Read(unsigned char *buff, int size) {
+
+bool FtdiInterface::Write(ola::io::ByteString *packet) {
+  int bytesWritten = ftdi_write_data(
+                       &m_handle,
+                       const_cast<unsigned char *>(packet->data()),
+                       packet->size());
+
+  int size = packet->size();
+  if (bytesWritten < 0) {
+    OLA_WARN << m_parent->Description() << " "
+             << ftdi_get_error_string(&m_handle);
+    return false;
+  } else if (bytesWritten == static_cast<int>(packet->size())) {
+    return true;
+  } else {
+    OLA_WARN << "Bytes Written: " << bytesWritten << " Packet Size: " << size;
+    return false;
+  }
+}
+
+int FtdiInterface::Read(unsigned char *buff, int size) {
   int read = ftdi_read_data(&m_handle, buff, size);
+  OLA_INFO << "FtdiRead";
   if (read <= 0) {
     OLA_WARN << m_parent->Description() << " "
              << ftdi_get_error_string(&m_handle);
-    return false;
+    return read;
```

Comment:
The return doesn't need to be in the if now does it, just the log line in there?

Reply:
Indeed